core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareHandler.java

```diff
@@ -283,6 +283,7 @@ private void setFinalResult(Prepared prepared) {
       ThrottledAdminRequestHandler<ByteBuffer> handler =
           ThrottledAdminRequestHandler.prepare(
               channel,
+              false,
```

Comment:
why we don't need to pre-acquire here?

Reply:
Because the channel was obtained from `session.getChannel`, which already pre-acquires (as part of selecting the channel from the pool, see `ChannelSet.next()`).

That's the kind of complexity we have to deal with after this change, unfortunately.